### PR TITLE
logical nand

### PR DESCRIPTION
**Pull request by RiteshKB**

### CODE REVIEW

File: symengine/tests/logic/test_logic.cpp [hunk 1]
```diff
@@ -176,6 +177,18 @@ TEST_CASE("And, Or : Basic", "[basic]")
...
+TEST_CASE("Nand : Basic", "[basic]")
+{
+    set_boolean e;
+    REQUIRE(eq(*logical_nand(e), *boolFalse));
+
+    REQUIRE(eq(*logical_nand({boolTrue}), *boolFalse));
+    REQUIRE(eq(*logical_nand({boolFalse}), *boolTrue));
+
+    REQUIRE(eq(*logical_nand({boolTrue, boolFalse}), *boolTrue));
+
```
Comment (nishnik) [inline symengine/tests/logic/test_logic.cpp]: This line is not needed

**Comment by isuruf** [comment]
What does SymPy do for `Nand`?

**Comment by RiteshKB** [comment]
@isuruf  i have added some tests with the help of sympy.

**Comment by isuruf** [comment]
Looking at the PR, the class `Nand` is not used anywhere, I would just remove it.

**Comment by isuruf** [comment]
I looked at what SymPy does and it immediately returns `Not(And(x, y))` instead of `Nand(x, y)`.
I think SymPy does this because `Not(And(x, y))` gives more information and will be helpful in automatic simplifying. So, a separate class `Nand` is not needed at all.

**Comment by isuruf** [comment]
@RiteshKB, your system time seems to be wrong. Can you fix it and edit the commit?

```
$ git show --format=fuller
commit 32f31a7a8e0e9e2af92f4525ad7fa4426e4102d4
Author:     comperror <babarwal.ritesh@gmail.com>
AuthorDate: Tue Oct 25 07:27:08 2016 +0530
Commit:     comperror <babarwal.ritesh@gmail.com>
CommitDate: Tue Oct 25 19:51:27 2016 +0530
```

**Comment by RiteshKB** [comment]
@isuruf  what else need to be done?

**Comment by isuruf** [comment]
Looks good

**Pull request merged**